mimic/rest/rackconnect_v3_api.py

```diff
@@ -0,0 +1,436 @@
+# -*- test-case-name: mimic.test.test_rackconnect_v3 -*-
+
+"""
+API mock for the Rackspace RackConnect v3 API, which is documented at:
+http://docs.rcv3.apiary.io/
+"""
+from collections import defaultdict
+import json
+from uuid import uuid4, UUID
+
+from characteristic import attributes, Attribute
+from six import text_type
+
+from twisted.plugin import IPlugin
+from twisted.web.http import NOT_FOUND, NOT_IMPLEMENTED
+from twisted.web.server import Request
+from zope.interface import implementer
+
+from mimic.catalog import Entry
+from mimic.catalog import Endpoint
+from mimic.imimic import IAPIMock
+from mimic.rest.mimicapp import MimicApp
+from mimic.util.helper import (attribute_names, random_ipv4,
+                               seconds_to_timestamp)
+
+
+Request.defaultContentType = 'application/json'
+timestamp_format = '%Y-%m-%dT%H:%M:%SZ'
+
+
+@implementer(IAPIMock, IPlugin)
+class RackConnectV3(object):
+    """
+    API mock object for RackConnect V3.
+
+    :ivar regions: The regions for which endpoints should be produced
+    :type regions: ``iterable``
+
+    :ivar int default_lbs: The number of default load balancers to be created
+        per tenant per region
+    """
+
+    def __init__(self, regions=("ORD",), default_lbs=1):
+        """
+        Construct a :class:`RackConnectV3` object.
+        """
+        self.regions = regions
+        self.default_lbs = default_lbs
+
+    def catalog_entries(self, tenant_id):
+        """
+        Catalog entry for RackConnect V3 endpoints.
+        """
+        return [
+            Entry(tenant_id, "rax:rackconnect", "rackconnect", [
+                Endpoint(tenant_id, region, text_type(uuid4()), prefix="v3")
+                for region in self.regions
+            ])
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Return an IResource implementing a public RackConnect V3 region
+        endpoint.
+        """
+        return RackConnectV3Region(
+            iapi=self,
+            uri_prefix=uri_prefix,
+            session_store=session_store,
+            region_name=region,
+            default_lbs=self.default_lbs).app.resource()
+
+
+lb_pool_attrs = [
+    Attribute("id", default_factory=lambda: text_type(uuid4())),
+    Attribute("name", default_value=u"default", instance_of=text_type),
+    Attribute("port", default_value=80, instance_of=int),
+    Attribute("status", default_value=u"ACTIVE", instance_of=text_type),
+    Attribute("status_detail", default_value=None),
+    Attribute("virtual_ip", default_factory=random_ipv4),
+    Attribute('nodes', default_factory=list, instance_of=list)]
+
+
+@attributes(lb_pool_attrs, apply_with_cmp=False)
+class LoadBalancerPool(object):
+    """
+    Represents a Load Balancer Pool, which cannot be created via the API.
```

Comment:
Nitpick: stating its inability to be created from the API doesn't seem like a primary description of what it does do, but it is on the first line regardless.